tests/core/pyspec/eth2spec/test/altair/transition/test_activation.py

```diff
@@ -0,0 +1,228 @@
+    assert not spec.is_in_inactivity_leak(state)
+    assert spec.get_current_epoch(state) < fork_epoch
+
+    yield "pre", state
+
+    # irregular state transition to handle fork:
+    blocks = []
+    state, block = do_altair_fork(state, spec, post_spec, fork_epoch)
+    blocks.append(post_tag(block))
+
+    # ensure that some of the current sync committee members are exiting
+    exited_pubkeys = [state.validators[index].pubkey for index in exited_indices]
+    assert any(set(exited_pubkeys).intersection(list(state.current_sync_committee.pubkeys)))
+
+    # continue regular state transition with new spec into next epoch
+    transition_to_next_epoch_and_append_blocks(post_spec, state, post_tag, blocks)
+
+    # check state
+    for index in exited_indices:
+        validator = state.validators[index]
+        assert not validator.slashed
+        assert post_spec.is_active_validator(validator, post_spec.get_current_epoch(state))
+    assert not post_spec.is_in_inactivity_leak(state)
+
+    yield "blocks", blocks
+    yield "post", state
+
+
+@fork_transition_test(PHASE0, ALTAIR, fork_epoch=2)
+def test_transition_with_one_fourth_exiting_validators_exit_at_fork(
+        state, fork_epoch, spec, post_spec, pre_tag, post_tag):
+    """
+    1/4 exiting but still active validators at the fork transition.
+    """
+    exited_indices = set_validators_exit_epoch(spec, state, exit_epoch=2, rng=random.Random(5566), fraction=0.25)
+
+    transition_until_fork(spec, state, fork_epoch)
+
+    # check pre state
+    assert len(exited_indices) > 0
+    for index in exited_indices:
+        validator = state.validators[index]
+        assert not validator.slashed
+        assert fork_epoch == validator.exit_epoch < spec.FAR_FUTURE_EPOCH
+        assert spec.is_active_validator(validator, spec.get_current_epoch(state))
+    assert not spec.is_in_inactivity_leak(state)
+    assert spec.get_current_epoch(state) < fork_epoch
+
+    yield "pre", state
+
+    # irregular state transition to handle fork:
+    blocks = []
+    state, block = do_altair_fork(state, spec, post_spec, fork_epoch)
+    blocks.append(post_tag(block))
+
+    # check post transition state
+    for index in exited_indices:
+        validator = state.validators[index]
+        assert not validator.slashed
+        assert not post_spec.is_active_validator(validator, post_spec.get_current_epoch(state))
+    assert not post_spec.is_in_inactivity_leak(state)
+
+    # ensure that none of the current sync committee members are exited validators
+    exited_pubkeys = [state.validators[index].pubkey for index in exited_indices]
+    assert not any(set(exited_pubkeys).intersection(list(state.current_sync_committee.pubkeys)))
+
+    # continue regular state transition with new spec into next epoch
+    transition_to_next_epoch_and_append_blocks(post_spec, state, post_tag, blocks)
+
+    yield "blocks", blocks
+    yield "post", state
+
+
+@fork_transition_test(PHASE0, ALTAIR, fork_epoch=260)
```

Comment:
this is outside the scope of this PR but it would be nice if `fork_epoch` could take either a number for the actual epoch *or* a `Callable` that accepts the pre and post spec and then lets the user compute an epoch...

the underlying issue here is that `fork_epoch` is hard-coded to exceed the `SHARD_COMMITTEE_PERIOD`

just making a note for now